### ISSUE TITLE
use salt api for minion key accept

### ISSUE THREAD

**Issue by vkhatri** [shortdudey123/chef-salt#34]
Currently, this cookbook uses Ohai salt and node attributes to auto accept the minion key.

I am working on PR which uses `salt-api` to accept the minion key. Below are the PR details.

* add new attribute `default['salt']['key_accept_method']`
  - `default['salt']['key_accept_method'] = 'pub_key_sync'` # options: pub_key_sync, api_key_accept

* configure ohai only if `node['salt']['key_accept_method'] == 'pub_key_sync'` as much as possible across the recipes

* make API call to master to accept minion key `only_if` 
  - minion key has not been already accepted
  - `node['salt']['key_accept_method'] == 'api_key_accept'`
  - print warning message if failed to accept

* configure master with an `external_auth pam` user and password by default, which is optional and controlled by attribute `default['salt']['master']['api']['enable_default_user']`
  - check if `default['salt']['master']['api']['enable'] == true`
  - check if `default['salt']['master']['api']['enable_default_user'] == true`
  - add linux user `default['salt']['master']['api']['default_user_name'] & default['salt']['master']['api']['default_user_password']`
  - add `external_auth pam` user config for `default['salt']['master']['api']['default_user']` with permissions `*, @wheel, @runner` to `/etc/salt/master`
```yaml
external_auth:
  pam:
    default-user:
      - .*
      - '@wheel'
      - '@runner'
```

* add attributes for `recipes/minion` to make API call to master
```ruby
default['salt']['minion']['api'] = {
  'host' => 'localhost',
  'port' => 8000,
  'username' => 'default-user',
  'password' => 'default-passwd',
  'eauth' => 'pam',
  'use_ssl' => true,
  'verify' => false
}
```

**Comment by shortdudey123** [comment]
`username` and `password` should probably come from an encrypted data bag instead of a node attribute for security reasons

**Comment by vkhatri** [comment]
agreed, but I would like to keep the data bag optional. I can submit another PR to lookup data bag (encrypted) if this looks ok to merge.

**Comment by vkhatri** [comment]
never mind my last comment, i will use the data bag.

**Comment by vkhatri** [comment]
@shortdudey123 Do you still want to keep the ohai + node attributes for minion key accept?

The PR I have (will submit working, require base) tested does not rely on node attribute or ohai etc.

**Comment by shortdudey123** [comment]
yes, using the API should be optional and controlled via an attribute
